torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1756,6 +1786,28 @@ def _make_tensor_helper(shape, low=None, high=None):
                        SkipInfo('TestCommon', 'test_variant_consistency_jit',
                                 device_type='cuda', dtypes=[torch.float16]),
                    )),
+    OpInfo('cumprod',
+           dtypes=all_types_and_complex_and(torch.bool),
+           dtypesIfCUDA=all_types_and_complex_and(torch.bool, torch.float16),
+           test_inplace_grad=False,
+           skips=(
+               # Reference: https://github.com/pytorch/pytorch/issues/53360
+               # For integer inputs,
+               # inplace variant preserves dtype of `self` while method variant
+               # always promotes it to torch.long.
+               # >>> t = torch.randint(2, 10, (3, 2), dtype=torch.int8)
+               # >>> t.cumprod(0).dtype
+               # torch.int64
+               # >>> t.cumprod_(0).dtype
+               # torch.int8
+               SkipInfo('TestCommon', 'test_variant_consistency_eager',
+                        dtypes=[torch.bool, torch.uint8, torch.int8, torch.int16, torch.int32]),
+               SkipInfo('TestCommon', 'test_variant_consistency_jit',
+                        dtypes=[torch.bool, torch.float16]),
+               SkipInfo('TestCommon', 'test_out',
+                        dtypes=[torch.float32]),
```

Comment:
@mruberry , could you please tell whether it is something expect? I hit very weird error messages if these tests are not skipped, that is:
`consistency_jit_float16` causes `allclose` to fail, not matter what  precision I set.
`test_out` for `float32` fails with an error message "User warning not triggered".

Reply:
The first test is OK to skip and I have a PR coming that will fix and remove the skip, so no worries.

The second error is probably real... looking...

Reply:
The error appears to be real. torch.prod() does not exhibit the correct behavior when given an out tensor with an incorrect size. It's OK to keep the skips but we should file an issue and comment a link to the issue here. We should fix its out= behavior in the future.

Reply:
Strange that it only manifests itself with `float32`, which is weird.

Reply:
It's just because the test only runs in float32.